File: app/Policies/OperationLogPolicy.php
<?php

namespace App\Policies;

use App\Models\OperationLog;
use App\Models\User;

class OperationLogPolicy
{
    public function viewAny(User $user): bool
    {
        return $user->isAdmin();
    }

    public function view(User $user, OperationLog $log): bool
    {
        return $user->isAdmin();
    }

    public function create(User $user): bool
    {
        return false;
    }

    public function update(User $user, OperationLog $log): bool
    {
        return false;
    }

    public function delete(User $user, OperationLog $log): bool
    {
        return false;
    }
}

File: app/policies/operation_log_policy.rb
class OperationLogPolicy < ApplicationPolicy
  def index?
    user.admin? || user.captain? || user.driver?
  end

  def show?
    user.admin? || user.captain? || record.user_id == user.id
  end

  def create?
    user.admin?
  end

  def new?
    create?
  end

  def update?
    user.admin?
  end

  def edit?
    update?
  end

  def destroy?
    user.admin?
  end

  class Scope < Scope
    def resolve
      if user.admin? || user.captain?
        scope.all
      else
        scope.where(user_id: user.id)
      end
    end
  end
end
